todo_app/app.py

```diff
@@ -8,7 +11,46 @@
 
 @app.route('/')
 def index():
-    return 'Hello World!'
+   items = get_items()
+   sorted_items = sorted(items, key = itemgetter('status'), reverse=True)
+   return render_template('index.html', title='To Do App', items=sorted_items)
+
+
+@app.route('/items/add', methods = ['POST', 'GET'])
+def add_items():
+   if request.method == 'POST':
+      form_data = request.form
+      task = form_data["title"]
+      add_item(task)
+        
+   return redirect('/')
+
+
+@app.route('/items/complete', methods = ['POST', 'GET'])
+def complete_item():
+   if request.method == 'POST':
+      form_data = request.form
+      task_id = int(form_data["id"])
+      existing_item = get_item(task_id)
+      if (existing_item != None) and (existing_item['status'] != 'Completed'):
+         updated_item = { 'id': task_id, 'status': 'Completed', 'title': existing_item['title'] }
+         save_item(updated_item)
+         return redirect('/')
+
+   return redirect('/')
+
+
+@app.route('/items/remove', methods = ['POST', 'GET'])
+def remove_items():
+   if request.method == 'POST':
+      form_data = request.form
+      task_id = int(form_data["id"])
+      existing_item = get_item(task_id)
+      if (existing_item != None):
```

Comment:
You could move these two lines of logic into the remove_item function (maybe rename it remove_item_by_id if you want)